packages/contracts-bedrock/contracts/dispute/AttestationDisputeGame.sol

```diff
@@ -0,0 +1,287 @@
+            // Store the L2 block number that the `rootClaim` commits to.
+            mstore(add(ptr, 0x40), _l2BlockNumber)
+
+            disputeStructHash := keccak256(ptr, 0x60)
+        }
+
+        _typedDataHash = Hash.wrap(_hashTypedData(Hash.unwrap(disputeStructHash)));
+    }
+
+    /**
+     * @notice Initializes the contract.
+     */
+    function initialize() external initializer {
+        createdAt = Timestamp.wrap(uint64(block.timestamp));
+        frozenAttestorSet = systemConfig.attestorSet();
+        frozenSignatureThreshold = systemConfig.attestationThreshold();
+    }
+
+    /**
+     * @notice Returns the semantic version of the DisputeGame contract.
+     * @dev Current version: 0.0.1
+     * @return The semantic version of the DisputeGame contract.
+     */
+    function version() external pure override returns (string memory) {
+        assembly {
+            // Store the pointer to the string
+            mstore(returndatasize(), 0x20)
+            // Store the version ("0.0.1")
+            // len |   "0.0.1"
+            // 0x05|302E302E31
+            mstore(0x25, 0x05302E302E31)
+            // Return the semantic version of the contract
+            return(returndatasize(), 0x60)
+        }
+    }
+
+    /**
+     * @notice If all necessary information has been gathered, this function should mark the game
+     *         status as either `CHALLENGER_WINS` or `DEFENDER_WINS` and return the status of
+     *         the resolved game. It is at this stage that the bonds should be awarded to the
+     *         necessary parties.
+     * @dev May only be called if the `status` is `IN_PROGRESS`.
+     * @return _status The status of the resolved game.
+     */
+    function resolve() public returns (GameStatus _status) {
+        // Set the status as `CHALLENGER_WINS`.
+        status = GameStatus.CHALLENGER_WINS;
+
+        // Delete all outputs from [l2BlockNumber, currentL2BlockNumber]
+        l2OutputOracle.deleteL2Outputs(l2BlockNumber());
+
+        // Request the `BondManager` to distribute the faulty output bond to the attestors.
+        uint256 _l2BlockNumber = l2BlockNumber();
+        bondManager.seizeAndSplit(keccak256(abi.encode(_l2BlockNumber)), attestationSubmitters);
+
+        return status;
+    }
+
+    /**
+     * @notice Returns the type of proof system being used for the AttestationDisputeGame.
+     * @dev The reference impl should be entirely different depending on the type (fault, validity)
+     *      i.e. The game type should indicate the security model.
+     * @return _gameType The type of proof system being used.
+     */
+    function gameType() public pure override returns (GameType) {
+        return GameType.ATTESTATION;
+    }
+
+    /**
+     * @notice Overrides the EIP712 domain information
+     * @return _name The name of the domain.
+     * @return _version The version of the domain.
+     */
+    function _domainNameAndVersion()
```

Comment:
What do you all think about using this eip for retrieving the eip 712 domain?
https://ethereum-magicians.org/t/eip-5267-retrieval-of-eip-712-domain/9951/3

Reply:
Resolving since solady gives eip 5267 to us for free: https://github.com/Vectorized/solady/blob/main/src/utils/EIP712.sol#L119-L144